### PR TITLE
Task/bws54 workers table

### PR DESCRIPTION
**Pull request by Oleg-feriss**

### CODE REVIEW

File: backend/src/api/api.ts [hunk 1]
```diff
@@ -26,6 +27,12 @@ const initApi: FastifyPluginAsync = async (fastify) => {
     },
     prefix: ApiPath.TENANTS,
   });
+  fastify.register(initWorkersApi, {
+    services: {
+      worker,
+    },
+    prefix: ApiPath.WORKERS,
```
Comment (what1s1ove) [inline backend/src/api/api.ts]:
```suggestion
    prefix: ApiPath.IAM,
```

File: shared/src/common/enums/api/api-path.enum.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 enum ApiPath {
   MASTERS = '/masters',
   TENANTS = '/tenants',
+  WORKERS = '/workers',
```
Comment (what1s1ove) [inline shared/src/common/enums/api/api-path.enum.ts]:
```suggestion
  EAM = '/eam',
```

File: shared/src/common/enums/api/workers-api-path.enum.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+enum WorkersApiPath {
+  EAM = '/eam',
+}
```
Comment (what1s1ove) [inline shared/src/common/enums/api/workers-api-path.enum.ts]:
```suggestion
enum EAMApiPath {
  ROOT = '/'
  WORKERS = '/workers',
}
```

File: backend/src/api/api.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import { FastifyPluginAsync } from 'fastify';
 import { ApiPath } from '~/common/enums/enums';
-import { master, token, tenant } from '~/services/services';
+import { master, token, tenant, worker } from '~/services/services';
 import { initMastersApi } from './masters/masters.api';
 import { initTenantsApi } from './tenants/tenants.api';
+import { initWorkersApi } from './workers/workers.api';
```
Comment (what1s1ove) [inline backend/src/api/api.ts]:
```suggestion
import { initEAMApi } from './eam/eam.api';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/api.ts (hunk 1 above)
Comment (Yaroslaww-1) [reply backend/src/api/api.ts]: I think its better `prefix: ApiPath.EAM,`

File: backend/src/services/worker/worker.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+class Worker {
+  #workerRepository: typeof workerRep;
+
+  constructor({ workerRepository }: Constructor) {
+    this.#workerRepository = workerRepository;
+  }
+
+  async getAll(): Promise<EAMWorkerDto[]> {
```
Comment (Yaroslaww-1) [inline backend/src/services/worker/worker.service.ts]: `EAMWorkerDto` => `EAMWorkerGetAllResponseDto`;
`Promise<EAMWorkerDto[]>` => `Promise<EAMWorkerGetAllResponseDto>`
`type EAMWorkerGetAllResponseDto = {
  items: ...worker info []
}`

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: backend/src/services/worker/worker.service.ts [hunk 6]
```diff
@@ -80,6 +81,17 @@ class Worker {
...
+
+  async getAll(): Promise<EAMWorkerGetAllResponseDto> {
+    const workers = await this.#workerRepository.getAll();
+
+    const items = workers.map((m) => ({
+      id: m.id,
+      name: m.name,
+    }));
+
+    return { items };
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.service.ts]:
```suggestion
    return workers.map((m) => ({
      id: m.id,
      name: m.name,
    }));
```

Comment (what1s1ove) [reply backend/src/services/worker/worker.service.ts]: don't forget to remove all unnecessary types. Ex `EAMWorkerGetAllResponseDto `

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/worker/worker.service.ts [hunk 7]
```diff
@@ -80,6 +81,15 @@ class Worker {
 
     return await this.#workerRepository.create(worker);
   }
+
+  async getAll(): Promise<EAMWorkerGetAllItemResponseDto[]> {
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.service.ts]: five a better name for type something like `EAMWorkerResponseDto `

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/worker/worker.service.ts (hunk 7 above)
Comment (Yaroslaww-1) [reply backend/src/services/worker/worker.service.ts]:
```
---
type EAMWorkerGetAllItemResponseDto = {
  id: string;
  name: string;
};

type EAMWorkerGetAllResponseDto = {
  items: EAMWorkerGetAllItemResponseDto[];
};
----

async getAll(): Promise<EAMWorkerGetAllResponseDto> {
   const workers = await this.#workerRepository.getAll();
   return {
      items: workers.map((m) => ({
        id: m.id,
        name: m.name,
      }))
   } 
}

```

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

**Pull request merged**